docs/docs/reference/05-oauth-providers/mattermost.md

```diff
@@ -0,0 +1,36 @@
+---
+id: mattermost
+title: Mattermost
+---
+
+## Documentation
+
+https://developers.mattermost.com/integrate/apps/authentication/oauth2
+
+## Configuration
+
+http://my-cool-server.cloud.mattermost.com/mycoolteam/integrations/oauth2-apps
+
+## Options
+
+The **Mattermost provider** comes with a set of default options:
+
+- [Mattermost Provider options](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/mattermost.ts)
```

Comment:
the `/packages/next-auth/src/providers/mattermost.ts` part would make it a dead link, so I guess ill have to also place the new provider there aswell? ill wait for clarification :)